masq_lib/src/percentage.rs

```diff
@@ -51,174 +29,110 @@ macro_rules! impl_percentage_integer {
-                $significant_digits
-                    .$method_add_or_sub(&N::try_from(1).expect("Each type has 1"))
-                    .expect($msg_in_expect)
-            };
-        }
-
-        match desired_rounding {
-            RoundingTo::BiggerPositive => {
-                adjust_num!(significant_digits_only, checked_add, "Addition failed")
-            }
-            RoundingTo::BiggerNegative => {
-                adjust_num!(significant_digits_only, checked_sub, "Subtraction failed")
-            }
-            RoundingTo::SmallerNegative | RoundingTo::SmallerPositive => significant_digits_only,
-        }
-    }
-
-    fn should_be_rounded_to<N>(num: N, divisor: N) -> RoundingTo
-    where
-        N: PercentageInteger,
-        <N as TryFrom<i8>>::Error: Debug,
-    {
-        let least_significant_digits: N = num % divisor;
-        let is_signed = num < N::try_from(0).expect("Each type has 0");
-        let divider = N::try_from(50).expect("Each type has 50");
-        let abs_of_significant_digits =
-            Self::abs_of_least_significant_digits(least_significant_digits, is_signed);
-        let is_minor = abs_of_significant_digits < divider;
-        match (is_signed, is_minor) {
-            (false, true) => RoundingTo::SmallerPositive,
-            (false, false) => RoundingTo::BiggerPositive,
-            (true, true) => RoundingTo::SmallerNegative,
-            (true, false) => RoundingTo::BiggerNegative,
-        }
-    }
-
-    fn abs_of_least_significant_digits<N>(least_significant_digits: N, is_signed: bool) -> N
-    where
-        N: TryFrom<i8> + CheckedMul,
-        <N as TryFrom<i8>>::Error: Debug,
-    {
-        if is_signed {
+    fn __abs(num: N, is_negative: bool) -> N {
+        if is_negative {
             N::try_from(-1)
                 .expect("Negative 1 must be possible for a confirmed signed integer")
-                .checked_mul(&least_significant_digits)
-                .expect("Must be possible in these low values")
+                .checked_mul(&num)
+                .expect("Must be possible for these low values")
         } else {
-            least_significant_digits
+            num
         }
     }
 
-    pub fn add_percent_to<N>(&self, num: N) -> N
-    where
-        N: PercentageInteger,
-        <N as TryFrom<i8>>::Error: Debug,
-        i16: TryFrom<N>,
-        <i16 as TryFrom<N>>::Error: Debug,
-    {
-        let to_add = self.of(num);
+    // This function helps to correct the last digit of the resulting integer to be as close
+    // as possible to the hypothetical fractional number, if we could go beyond the decimal point.
+    fn __derive_rounding_increment(remainder: N) -> N {
+        let is_negative = remainder < N::try_from(0).expect("Each integer has 0");
+        let is_minor =
+            Self::__abs(remainder, is_negative) < N::try_from(50).expect("Each integer has 50");
```

Comment:
This part is hard to understand, why's that? Specifically the number 50, it's origin.